src/EnergyPlus/Timer.cc

```diff
@@ -0,0 +1,112 @@
+//     provided with the distribution.
+//
+// (3) Neither the name of the University of California, Lawrence Berkeley National Laboratory,
+//     the University of Illinois, U.S. Dept. of Energy nor the names of its contributors may be
+//     used to endorse or promote products derived from this software without specific prior
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <EnergyPlus/Timer.hh>
+
+#include <fmt/format.h>
+#ifndef NDEBUG
+#include <stdexcept>
+#endif
+
+namespace EnergyPlus {
+
+void Timer::tick()
+{
+    m_end = TimePointType{};
+    m_start = ClockType::now();
+}
+
+void Timer::tock()
+{
+#ifndef NDEBUG
+    if (m_start == TimePointType{}) {
+        throw std::runtime_error("Timer was not started");
+    }
+#endif
+    m_end = ClockType::now();
+    m_duration = m_duration + std::chrono::duration_cast<DurationType>(m_end - m_start);
+}
+
+Timer::DurationType Timer::duration() const
+{
+#ifndef NDEBUG
+    if (m_end == TimePointType{}) {
+        throw std::runtime_error("Timer was not stopped");
+    }
+#endif
+    return m_duration;
+}
+
+std::string Timer::formatAsHourMinSecs() const
+{
+    // We're working with ms as base.
+    // TODO: do we want to report elapsed time of 25 hours as 1 day 1hr 0min 0sec or 25hr 0min 0sec?
```

Comment:
@Myoldmopar forgot to flag this
Do we want 1 days 1 hr xxx or 2hr

Reply:
I do not care...  Possibly 1 day 1 hr ??  I think either is fine though.  This would only matter in a corner case where `(runtime is **enormously** long) .and. (the user cant handle a 25 hour time stamp)` which should round down to zero percent of the EnergyPlus use cases.  Anyway, whatever is happening right now is fine.